pymeasure/display/widgets.py

```diff
@@ -229,6 +229,241 @@ def update_y_column(self, index):
+                item.update_img()
+        xlabel, xunits = self.parse_axis(x_axis)
+        self.plot.setLabel('bottom', xlabel, units=xunits, **self.LABEL_STYLE)
+        self.x_axis = x_axis
+        self.x_axis_changed.emit(x_axis) # JM: Not sure if this does anything now...
+        ylabel, yunits = self.parse_axis(y_axis)
+        self.plot.setLabel('left', ylabel, units=yunits, **self.LABEL_STYLE)
+        self.y_axis = y_axis
+        self.y_axis_changed.emit(y_axis) # JM: Not sure if this does anything now...
+        self.change_z_axis(z_axis)
+
+    def _setup_ui(self):
+        self.setAutoFillBackground(False)
+        self.setStyleSheet("background: #fff")
+        self.setFrameShape(QtGui.QFrame.StyledPanel)
+        self.setFrameShadow(QtGui.QFrame.Sunken)
+        self.setMidLineWidth(1)
+
+        vbox = QtGui.QVBoxLayout(self)
+
+        self.plot_widget = pg.PlotWidget(self, background='#ffffff') # JM: NOTE: This is just pulling a base pg PlotWidget, so no infinite regress!
+        self.coordinates = QtGui.QLabel(self)
+        self.coordinates.setMinimumSize(QtCore.QSize(0, 20))
+        self.coordinates.setStyleSheet("background: #fff")
+        self.coordinates.setText("")
+        self.coordinates.setAlignment(
+            QtCore.Qt.AlignRight | QtCore.Qt.AlignTrailing | QtCore.Qt.AlignVCenter)
+
+        vbox.addWidget(self.plot_widget)
+        vbox.addWidget(self.coordinates)
+        self.setLayout(vbox)
+
+        self.plot = self.plot_widget.getPlotItem()
+
+        self.crosshairs = Crosshairs(self.plot,
+                                     pen=pg.mkPen(color='#AAAAAA', style=QtCore.Qt.DashLine))
+        self.crosshairs.coordinates.connect(self.update_coordinates)
+
+        self.timer = QtCore.QTimer()
+        self.timer.timeout.connect(self.update_curves)
+        self.timer.timeout.connect(self.crosshairs.update)
+        self.timer.timeout.connect(self.updated)
+        self.timer.start(int(self.refresh_time * 1e3))
+
+    def update_coordinates(self, x, y):
+        self.coordinates.setText("(%g, %g)" % (x, y))
+
+    def update_curves(self):
+        for item in self.plot.items:
+            if isinstance(item, ResultsImage):
+                if self.check_status:
+                    if item.results.procedure.status == Procedure.RUNNING:
+                        item.update_img()
+                else:
+                    item.update()
+
+    def parse_axis(self, axis):
+        """ Returns the units of an axis by searching the string
+        """
+        units_pattern = "\((?P<units>\w+)\)"
+        try:
+            match = re.search(units_pattern, axis)
+        except TypeError:
+            match = None
+            
+        if match:
+            if 'units' in match.groupdict():
+                label = re.sub(units_pattern, '', axis)
+                return label, match.groupdict()['units']
+        else:
+            return axis, None
+
+    # def change_x_axis(self, axis):
```

Comment:
Please remove the commented code in this file.

Reply:
Done.